### ISSUE TITLE
ctc_loss value problem

### ISSUE THREAD

**Issue by xushenkun** [tensorflow/tensorflow#15086]
------------------------
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: Yes
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Windows 7 64bit
- **TensorFlow installed from (source or binary)**: binary
- **TensorFlow version (use command below)**: 1.4.0
- **Python version**: 3.6.0
- **Bazel version (if compiling from source)**: None
- **GCC/Compiler version (if compiling from source)**: None
- **CUDA/cuDNN version**: None
- **GPU model and memory**: None
- **Exact command to reproduce**: None

### Describe the problem
I used the source code below to calculate ctc loss. The logit and target should extremely have no loss between them. But the tf.nn.ctc_loss return 1.91309595. What's the meaning of the tf ctc_loss result? 
But I use my custom ctc loss function, which shows the loss is nearly zero.

### Source code / logs

import numpy as np
import tensorflow as tf
logit = np.array([[[0.00001, 0.9999, 0.00001, 0.00001], [0.9999, 0.00001, 0.00001, 0.00001], [0.00001, 0.00001, 0.9999, 0.00001]]])
logit = tf.constant(logit, dtype='float32')
target = tf.SparseTensor(indices=[[0, 0], [0, 2]], values=[1, 2], dense_shape=[1, 3])
seq = tf.constant(np.array([3]))
loss_tf = tf.nn.ctc_loss(target, logit, seq, time_major=False, ctc_merge_repeated=True)
sess = tf.InteractiveSession()
tf.global_variables_initializer().run()
print(loss_tf.eval())

**Comment by marcoleewow** [comment]
I have encountered the same issue, the built-in tensorflow CTC API cannot pass this simple test from Alex Graves paper.

https://gist.github.com/marcoleewow/d7dc5078a36927b3f13dbf64ac9905ad

**Comment by marcoleewow** [comment]
@xushenkun I have figured out the problem, the CTC loss performs softmax operation inside, so you should feed in the activations, not the probabilities after softmax.

**Comment by marcoleewow** [comment]
@xushenkun I have now used non-softmax tensor to feed into `ctc_loss` function, but I still cannot achieve 0 loss with your example. I think this is a tensorflow bug, please double check my code [here](https://gist.github.com/marcoleewow/037fc43733260300083aab0db2d7d9a8).

**Comment by xushenkun** [comment]
@marcoleewow Thanks for your remind, I forgot the softmax issue. And I think although these two kinds of targets mean same dense tensor value, tf'ctc_loss use the sparse tensor directly, which are different.
